core/src/main/java/org/infinispan/statetransfer/StateTransferLock.java

```diff
@@ -39,15 +43,35 @@
    // transaction data latch
    void notifyTransactionDataReceived(int topologyId);
 
-   void waitForTransactionData(int expectedTopologyId, long timeout, TimeUnit unit) throws InterruptedException;
+   CompletableFuture<Void> transactionDataFuture(int expectedTopologyId);
+
+   @Deprecated
+   default void waitForTransactionData(int expectedTopologyId, long timeout, TimeUnit unit) throws InterruptedException, TimeoutException {
```

Comment:
You've asked for a temporary replacement until we fix the usages. As the method should not be in the interface either, we should not force it in the implementation.